### PR TITLE
feat: Add support to get marketplace item by itemID+version tuple

### PR DESCRIPTION
**Pull request by malta895**
## What this PR is for?

Add support to get marketplace item by itemID+version tuple

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,11 +23,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Updated
 
-- modified `miactl marketplace delete` command to accept either the `objectId` or a `itemId`-`version` tuple that identifies the item to be deleted.
+- modified `miactl marketplace delete` command to accept either the `objectId` or a `companyId`-`itemId`-`version` tuple that identifies the item to be deleted.
+- modified `miactl marketplace get` command to accept either the `objectId` or a `companyId`-`itemId`-`version` tuple that identifies the item to be deleted.
```
Comment (pmarchini) [inline CHANGELOG.md:27]:
```suggestion
- modified `miactl marketplace get` command to accept either the `objectId` or a `companyId`-`itemId`-`version` tuple that identifies the item to be retrieved.
```

File: docs/30_commands.md [hunk 2]
```diff
@@ -331,10 +331,19 @@ Get a Marketplace item
 
 #### Synopsis
 
-Get a single Marketplace item by its ID
+Get a single Marketplace item
 
-```
-miactl marketplace get resource-id [flags]
+You need to specify either:
+- the itemId and the version, via the respective flags (recommended)
```
Comment (pmarchini) [inline docs/30_commands.md:337]:
```suggestion
- the itemId, the companyId and the version, via the respective flags (recommended)
```
We use to set it into the context but is needed if not provided via context

File: docs/30_commands.md [hunk 3]
```diff
@@ -346,14 +355,16 @@ Delete a Marketplace item
 Delete a single Marketplace item
 
 You need to specify either:
-- the itemId and the version, respectively  (recommended)
+- the itemId and the version, via the respective flags (recommended)
```
Comment (pmarchini) [inline docs/30_commands.md:358]: Do we need the complete tuple here?

File: internal/cmd/marketplace/get.go [hunk 4]
```diff
@@ -94,19 +113,26 @@ func getMarketplaceItemByID(client *client.APIClient, resourceID string) (*marke
...
+func getItemEncodedWithFormat(client *client.APIClient, objectID, companyID, itemID, version, outputFormat string) (string, error) {
+	var item *marketplace.Item
+	var err error
+	if objectID != "" {
+		item, err = getItemByObjectID(client, objectID)
+	} else {
+		if companyID == "" {
+			return "", marketplace.ErrMissingCompanyID
+		}
+		item, err = getItemByItemIDAndVersion(client, companyID, itemID, version)
```
Comment (pmarchini) [inline internal/cmd/marketplace/get.go:126]: What is supposed to happen if itemId or version is missing?

File: internal/cmd/marketplace/get_test.go [hunk 5]
```diff
@@ -40,121 +84,258 @@ func TestGetResourceCmd(t *testing.T) {
...
+			statusCode:          http.StatusNotFound,
+			expectError:         true,
+			outputFormat:        encoding.JSON,
+			expectedCalledCount: 1,
+
+			companyID: mockCompanyID,
+			itemID:    mockItemID,
+			version:   mockVersion,
+		},
+
```
Comment (pmarchini) [inline internal/cmd/marketplace/get_test.go:266]: space

File: internal/cmd/marketplace/get.go (hunk 4 above)
Comment (malta895) [reply internal/cmd/marketplace/get.go:126]: It's impossible they are missing at this point, the command flags are marked as required

File: docs/30_commands.md (hunk 2 above)
Comment (malta895) [reply docs/30_commands.md:337]: I wrote it as a foot note. Do you think it is also needed here?

File: docs/30_commands.md (hunk 3 above)
Comment (malta895) [reply docs/30_commands.md:358]: Yes, but also here is written as a foot note.

Comment (malta895) [reply docs/30_commands.md]: clarified

File: docs/30_commands.md [hunk 6]
```diff
@@ -331,10 +331,19 @@ Get a Marketplace item
...
-miactl marketplace get resource-id [flags]
+You need to specify either:
+- the companyId, itemId and version, via the respective flags (recommended). The company-id flag can be omitted if it is already set in the context.
+- the ObjectID of the item with the flag object-id
+
+Passing the ObjectID is expected only when dealing with deprecated Marketplace items missing the itemId and/or version fields.
+Otherwise, it is preferable to pass the tuple companyId-itemId-version.
+
+The companyID must be passed via the flag company-id when not set in the context.
+
```
Comment (malta895) [inline docs/30_commands.md:344]:
```suggestion
```

**Review by pmarchini — APPROVED** [review]

**Comment by pmarchini** [comment]
LGTM

**Pull request merged**